mysql.go

```diff
@@ -417,7 +421,32 @@ func (dialector Dialector) getSchemaBytesType(field *schema.Field) string {
 	return "longblob"
 }
 
+// autoRandomType
+// field.DataType MUST be `schema.Int` or `schema.Uint`
+// Judgement logic:
+// 1. Is PrimaryKey;
+// 2. Has default value;
+// 3. Default value is "auto_random()";
+// 4. IGNORE the field.Size, it MUST be bigint;
+// 5. CLEAR the default tag, and return true;
+// 6. Otherwise, return false.
+func autoRandomType(field *schema.Field) (bool, string) {
+	if field.PrimaryKey && field.HasDefaultValue &&
+		strings.ToLower(strings.TrimSpace(field.DefaultValue)) == AutoRandomTag {
+		field.HasDefaultValue = false
```

Comment:
Why do we need to set `HasDefaultValue` to false, it should be true?

Reply:
We used this tag `default:auto_random()` to implement the `auto_random` feature. But it is actually not a default value. It's a field like `auto_increment`.

Reply:
for `auto_increment` fields we also marked them as `HasDefaultValue`, e.g:

https://github.com/go-gorm/gorm/blob/d834dd60b715422dc2a900fb2744f9c278a9830f/schema/field.go#L113

Reply:
Ah, got it. I made a mistake.

Reply:
Fixed. PTAL again.